.dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM node:18.6-slim
+
+LABEL maintainer="DocNetwork <webteam@docnetwork.org>"
+
+ARG codeArtifactToken
+
+WORKDIR /dn-dev-availability-service
+
+COPY . /dn-dev-availability-service/
+
+
+RUN npm config set registry=https://dn-878788551012.d.codeartifact.us-east-1.amazonaws.com/npm/dn-npm-public/
```

Comment:
Totally optional but, you could refactor this to be a little cleaner and a little more flexible.

On line 11 add ```ARG registry=dn-878788551012.d.codeartifact.us-east-1.amazonaws.com```

Then on lines 12, 13, and 14 you can replace that string with $registry

by doing this you set one place to update the registry but also you can pass in the value with the --build-arg parameter from the command line.  I will leave it to you to decide if you want to include the whole path or not.